pkg/kubelet/kubelet_pods.go

```diff
@@ -1352,7 +1352,13 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 			s.Phase = pod.Status.Phase
 		}
 	}
-	kl.probeManager.UpdatePodStatus(pod.UID, s)
+	oldContainerReadiness := make(map[string]bool)
+	if p, found := kl.podManager.GetPodByName(pod.Namespace, pod.Name); found{
+		for _, c := range p.Status.ContainerStatuses{
+			oldContainerReadiness[c.ContainerID] = c.Ready
+		}
+	}
+	kl.probeManager.UpdatePodStatus(pod.UID, s, oldContainerReadiness)
```

Comment:
Could you update `s.Status.ContainerStatuses` to `oldContainerReadiness` directly? Then we do not need to add another argument `oldContainerReadiness`.

Reply:
Yes, it's simpler that way, let me handle it, thanks!